go/types/struct.go

```diff
@@ -195,13 +209,45 @@ func EscapeStructField(input string) string {
 		buf.WriteString(hs)
 		return buf.String()
 	}
+	output := EscapeFields(input, encode)
+	verifyFieldName(output)
+	return output
 
-	output := ""
-	pattern := headPattern
-	for _, ch := range input {
-		output += encode(string([]rune{ch}), pattern)
-		pattern = tailPattern
+}
+
+// IsValidStructFieldName returns whether the name is valid without as a field name in a struct.
```

Comment:
This comment doesn't read clearly - "...name is valid without as..."? "without encoding" maybe, not sure?